File: src/leetcode/python/greedy/PatchingArray_330.py
class PatchingArray_330(object):
    def minPatches(self, nums, n):
        """
        :type nums: List[int]
        :type n: int
        :rtype: int
        """
        miss, patch, index = 1, 0, 0
        length = len(nums)
        while miss <= n:
            if index < length and nums[index] <= miss:
                miss += nums[index]
                index += 1
            else:
                miss += miss
                patch += 1
        return patch

File: src/leetcode/java/greedy/PatchingArray_330.java
package leetcode.java.greedy;

public class PatchingArray_330 {
  public int minPatches(int[] nums, int n) {
    int patch = 0;
    int miss = 1, index = 0, len = nums.length;
    while (miss > 0 && miss <= n) {
      if (index < len && nums[index] <= miss) {
        miss += nums[index++];
      } else {
        miss += miss;
        patch++;
      }
    }
    return patch;
  }
}
